### ISSUE TITLE
Inline completion preselection of intellisense items doesn't work if it includes a closing brace which is in the document.

### ISSUE THREAD

**Issue by vivlimmsft** [microsoft/vscode#135440]
Does this issue occur when all extensions are disabled?: Yes/**No** (because I would not get any inline completions)

- VS Code Version: 1.62.0
- OS Version: win11 22000.258

Steps to Reproduce:

1. Trigger an inline completion which meets all of the following conditions
* Inline completion matches an item in the intellisense list
* Inline completion contains a closing parenthesis
* The closing parenthesis already exists in the document.

(thanks @lostintangent for providing this repro gif, using [this file](https://github.com/lostintangent/gitdoc/blob/8e11a8dfb8bab01477f356054d6bce8bf3d69dbb/src/watcher.ts#L60))
![c6766371-8c43-48a4-a660-3bbdc5a9650d](https://user-images.githubusercontent.com/27872315/138016185-219469c6-5fdb-4bf5-a592-f413f35e10c4.gif)

2. Trigger intellisense (i.e. by ctrl+space). The matching item is *not* preselected.

I dug into this a bit and found that the ranges that the inline completion item and intellisense item are applying to differ.

At this location:
![image](https://user-images.githubusercontent.com/27872315/138016474-3ec6374f-b4ae-4fce-898d-b51a3f8ae35d.png)
I saw these watch values:
![image](https://user-images.githubusercontent.com/27872315/138016511-b1314c8e-45ce-4a55-ae03-f7658a0517b2.png)

the second condition, that the beginning of the text matches, is true. the condition that the range matches is false, because the endColumn is 53 and 54.

![image](https://user-images.githubusercontent.com/27872315/138016662-a0f37da2-2644-4bd4-9a2c-65daa224aaf9.png)
the 53rd character on this line is the trailing )
![image](https://user-images.githubusercontent.com/27872315/138016681-6c67dc3c-076f-4949-9b18-f755bb23041d.png)
![image](https://user-images.githubusercontent.com/27872315/138016684-0b98d4e4-d64f-4f23-8c87-a8306cbd1952.png)

so my guess of why they differ is that the inline completion range overlaps character 53 because the completion *includes the closing paren* which is *already in the document*.

@hediet , can the condition used to match inline suggestions to intellisense items be relaxed here, perhaps only requiring the `startLineNumber`s and `startColumn`s to match?

( cc @juihanamshet1 and @ayim )

**Comment by vivlimmsft** [comment]
@hediet , sorry for the >2 week delay in validating the fix, but I found that it didn't work for me.

I made this change locally which fixed it:
![image](https://user-images.githubusercontent.com/27872315/141361977-dd39b2a9-f814-4578-a024-50406543dcf4.png)

I assume the range to test should always be as long as or longer than the prefix that it may start with. Does that make sense?
Should there also be an inversion of `rangeToTest.endLineNumber < prefix.endLineNumber`?
